flink-runtime/src/main/scala/org/apache/flink/runtime/jobmanager/JobManager.scala

```diff
@@ -2041,6 +2044,18 @@ object JobManager {
         null
     }
 
+    val secureAuth = configuration.getBoolean(ConfigConstants.SECURITY_ENABLED,
+                                              ConfigConstants.DEFAULT_SECURITY_ENABLED)
+    if(secureAuth == true) {
+      val secureCookie = configuration.getString(ConfigConstants.SECURITY_COOKIE, null)
+      if(secureCookie == null) {
+        val message = "Missing " + ConfigConstants.SECURITY_COOKIE +
+                      " configuration in Flink configuration file"
+        LOG.error(message)
+        throw new RuntimeException(message)
+      }
+    }
```

Comment:
As I said before, this is duplicate code that can be encapsulated into a `SecurityUtils` static method.